### PR TITLE
dApps-Staking chain-extension - Reworked Errors

### PR DESCRIPTION
**Pull request by PierreOssun**
Before the chain extension returned Result<() DispatchError> but it was not possible to decode it on contract side (in a handy way).
The appropriate way is to write to buffer when a value needs to be returned.
When a Result should be returned it should be wrapped in a RetVal and decoded on the contract side. This way no scale decoding error can happen.

**Check list**
- [X] Updated Trait function Return type
- [X] Updated all return statements

### CODE REVIEW

**Review by shunsukew — APPROVED** [review]

File: chain-extensions/types/dapps-staking/src/lib.rs [hunk 1]
```diff
@@ -102,7 +106,7 @@ impl TryFrom<DispatchError> for DSError {
             Some("NominationTransferToSameContract") => {
                 Ok(DSError::NominationTransferToSameContract)
             }
-            _ => Err(DispatchError::Other("DappsStakingExtension: Unknown error")),
+            _ => Ok(DSError::UnknownError),
```
Comment (Maar-io) [inline chain-extensions/types/dapps-staking/src/lib.rs:109]: I think this should remain `DispatchError`, since the `try_from()` will fail and `execute_func()` will return `Result<RetVal, DispatchError>`

Comment (PierreOssun) [reply chain-extensions/types/dapps-staking/src/lib.rs:109]: `try_from()` will not fail. it will return `DSError::UnknownError` and will be decoded on contract side.
When your try to map DispatchError to DSError - it will map to an DSError in any case - so you can handle it in contract side. 
For example, if there is a new Error added in dapps-staking pallet that was not added to this TryForm impl - it will return UnknownError and we should update our DSError.
Still `execute_func()` can return an DispatchError if it fails on substrate level fucntions (ex: env().write) which will no be handled by the contract. Because query calls should not fail (only return a value not a Result). 
To Sum up:
- Any business error returned from dapps-staking pallet should be handled
- If the Err is not a business logic Err coming from dapps-staking it returns a Dispatch Error
- In a case where a business logic Err is returned from dapps-staking pallet is not mapped to DSError correctly it fails under UnknownError and should be immediatly added to DSError (this case should not happen actually as DSError should have exhaustive enum fiields of dapps-staking pallets errors - but we still cover it)

Comment (Maar-io) [reply chain-extensions/types/dapps-staking/src/lib.rs:109]: ok, I see, It will just be harder to debug in some cases since execute_func() will never fail.

**Review by Maar-io — APPROVED** [review]
LGTM

File: chain-extensions/types/dapps-staking/src/lib.rs (hunk 1 above)
Comment (PierreOssun) [reply chain-extensions/types/dapps-staking/src/lib.rs:109]: It will be easy to debug because it return a RetVal.
It can still send Err at every line where there is an `?` but it is Err outside of dApps-staking logic.
`try_from()` only return DsError not DispatchError anymore
